tests/format/format_test.py

```diff
@@ -23,6 +23,85 @@ def test_variable_default(var):
     assert '{:}'.format(var) == str(var)
 
 
+@pytest.mark.parametrize('s', ('^', ''))
+def test_variable_default_length_central(s):
+    var = sc.arange('x', 10)
+    assert '[0, 1, ..., 8, 9]' in f'{var:{s}}'
+    assert '[0, 1, ..., 8, 9]' in f'{var:{s}#4}'
+    assert '[0, 1, ..., 7, 8, 9]' in f'{var:{s}#5}'
+    assert '[0, 1, 2, ..., 7, 8, 9]' in f'{var:{s}#6}'
+
+    var = sc.arange('x', 4)
+    assert '[0, 1, 2, 3]' in f'{var:{s}}'
+    assert '[0, 1, 2, 3]' in f'{var:{s}#5}'
+    assert '[0, ..., 2, 3]' in f'{var:{s}#3}'
+    assert '[0, ..., 3]' in f'{var:{s}#2}'
+    assert '[..., 3]' in f'{var:{s}#1}'
+    assert '[...]' in f'{var:{s}#0}'
+
+    var = sc.arange('x', 0)
+    assert '[]' in f'{var:{s}}'
+    assert '[]' in f'{var:{s}#6}'
+    assert '[]' in f'{var:{s}#0}'
+
+    var = sc.scalar(5)
+    assert '[5]' in f'{var:{s}}'
+    assert '[5]' in f'{var:{s}#2}'
+    assert '[...]' in f'{var:{s}#0}'
```

Comment:
Can we avoid the `[]` for scalars? A 0-D numpy array also just prints a number.

Reply:
We can. But then we should change this in the C++ formatter, too. Don't know how easy that is...

Reply:
Done